python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -119,6 +120,26 @@ def __init__(
         )
 
 
+class Write(AbstractMap):
```

Comment:
@c21 @jianoaix Should `Write` derive from `AbstractMap`, or should it be a standalone logical operator like `Read`? https://github.com/ray-project/ray/blob/2248ea602fbf6c53db1c5afc58f8bd386a66e1de/python/ray/data/_internal/logical/operators/read_operator.py

Then we could avoid this awkwardness with `fn` and the like. I don't see a compelling reason to have `Write` derive from `AbstractMap`, on first pass.

Reply:
@clarkzinzow - I think if we make `Write` not `AbstractMap`, we need to add some extra code during planning ([current code](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/planner/plan_map_op.py#L47-L84)).

Would it break anything for operator fusion and zero-copy batching? If not, I am also in favor to make `Write` extend `LogicalOperator` directly, because `Read` is doing that, and we can add as a seprate `plan_write_op`.

Reply:
@c21 Yeah I think it's worth breaking out the separate planning bit, and it shouldn't break anything with operator fusion or zero-copy batching; it will actually line up better with the latter!

I have this bit refactored to make the dispatch easier to centralize beyond just the `AbstractMap` ops: https://github.com/ray-project/ray/pull/32178/files#diff-4caa8ddd8103dd8f8d6a3e8c1237aec4eaa168a81dc914ae83b4f6042d68a1da

Reply:
cool then +1 to make `Write` extend `LogicalOperator` directly.

Reply:
yeah I was feeling the `fn` was weird to have since it's not relevant for `Write`.